tools/checkstyle/suppressions.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suppressions PUBLIC "-//Puppy Crawl//DTD Suppressions 1.1//EN" "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+<suppressions>
+    <suppress files="build.properties" checks="AboutHtmlCheck" />
+    <suppress files=".classpath" checks="MavenPomderivedInClasspathCheck" />
+    <!-- These suppressions define which files to be suppressed for which checks. -->
+    <suppress files=".+[\\/]internal[\\/].+\.java" checks="JavadocType|JavadocVariable|JavadocMethod|JavadocFilterCheck|NullAnnotationsCheck"/>
+    <suppress files=".+DTO\.java" checks="JavadocType|JavadocVariable|JavadocMethod|JavadocFilterCheck" />
+    <suppress files=".+Impl\.java" checks="JavadocType|JavadocVariable|JavadocMethod|JavadocFilterCheck"/>
+
+    <!-- The WiFi led binding contains test packages that shouldn't be included in the binary build -->
+    <!-- The CometVisu UI contains files in the resources folder that are used only during the build -->
+    <suppress files=".+org.openhab.binding.wifiled.+|.+org.openhab.ui.cometvisu.+" checks="BuildPropertiesCheck"/>
+    <!-- Homematic and Tellstick bindings are creating and configuring things dynamically, they will log false positives for unused configuration -->
+    <!-- IO and Voice bundles have specific use cases - they use only the config .xml files and will log false positives as well -->
+    <suppress files=".+org.openhab.binding.homematic.+|.+org.openhab.binding.tellstick.+|.+org.openhab.io.+|.+org.openhab.voice.+" checks="EshInfXmlCheck"/>
```

Comment:
Disable the entire check for all bundles. A binding is allowed to have xml files that are only dynamically used as templates etc. I'm doing this in the yamaha binding for example.

Reply:
I think this is checking forconformance of files which is suppressed in special cases?

Reply:
Yes it checks if a config xml file is referenced from a thing xml file. But that is nonsense IMO because a binding developer is allowed to have unreferenced config xml files.